src/spanned.rs

```diff
@@ -120,17 +132,18 @@ where
         self.to_tokens(&mut tokens);
         let mut iter = tokens.into_iter();
         let mut span = match iter.next() {
-            Some(tt) => tt.span(),
+            Some(tt) => check_invalid_span(tt.span()),
             None => {
                 return Span::call_site();
             }
         };
         for tt in iter {
-            if let Some(joined) = span.join(tt.span()) {
+            if let Some(joined) = span.join(check_invalid_span(tt.span())) {
```

Comment:
Invalid spans need to be filtered out, not replaced with call_site. The behavior of Spanned::span should be to join all the valid spans in the input.

If there were no valid spans at all then we can return call_site.